File: BasicsWithJava/alertDialog/app/src/main/java/com/sumere/alertdialog/MainActivity.java
package com.sumere.alertdialog;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void save(View view){
        AlertDialog.Builder alert = new AlertDialog.Builder(this);
        alert.setTitle("Save");
        alert.setMessage("Are you sure?");
        alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                //Save
                Toast.makeText(MainActivity.this,"Toast Message\nSaved",Toast.LENGTH_LONG).show();
            }
        });
        alert.setNegativeButton("No",new DialogInterface.OnClickListener(){
            @Override
            public void onClick(DialogInterface dialogInterface,int i){
                Toast.makeText(MainActivity.this,"Toast Message\nNot saved",Toast.LENGTH_LONG).show();
            }
        });
        alert.show();
    }
}

File: AlertDialog/app/src/main/java/com/sumere/alertdialog/MainActivity.kt
package com.sumere.alertdialog

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // 1) Toast 2) Alert Dialog 3) Snackbar
        // Context = Information about application
    }

    fun save(view : View){
        //Toast.makeText(this@MainActivity,"Are You Sure?",Toast.LENGTH_LONG).show()
        val alert = AlertDialog.Builder(this@MainActivity)
        alert.setTitle("Save")
        alert.setMessage("Are You Sure?")
        alert.setPositiveButton("Yes",object : DialogInterface.OnClickListener{
            override fun onClick(p0: DialogInterface?, p1: Int) {
                Toast.makeText(this@MainActivity,"Saved!",Toast.LENGTH_LONG).show()
            }
        })
        alert.setNegativeButton("No",object:DialogInterface.OnClickListener{
            override fun onClick(p0: DialogInterface?, p1:Int){
                Toast.makeText(this@MainActivity,"Not Saved",Toast.LENGTH_LONG).show()
            }
        })
        alert.show()
    }
}